analyses/golang/matcher.go

```diff
@@ -0,0 +1,58 @@
+package golang
+
+import (
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+
+	"github.com/rs/zerolog/log"
+
+	"github.com/fabric8-analytics/cli-tools/analyses/driver"
+)
+
+var (
+	_ driver.StackAnalysisInterface = (*Matcher)(nil)
+)
+
+// Matcher is State Object for Golang
+type Matcher struct{}
+
+// Ecosystem implements driver.Matcher.
+func (*Matcher) Ecosystem() string { return "golang" }
+
+// DepsTreeFileName implements driver.Matcher.
+func (*Matcher) DepsTreeFileName() string { return "golist.json" }
+
+// GeneratorDependencyTree creates golist.json from go.mod
+func (m *Matcher) GeneratorDependencyTree(manifestFilePath string) string {
+	log.Debug().Msgf("Executing: Generate golist.json")
+	golang, err := exec.LookPath("go")
+	if err != nil {
+		log.Fatal().Err(err).Msgf("Please make sure Golang is installed. Hint: Check same by executing: go version \n")
+	}
+	manifestDir := filepath.Dir(manifestFilePath)
+	treePath, _ := filepath.Abs(filepath.Join(os.TempDir(), m.DepsTreeFileName()))
+	genRepo := "github.com/fabric8-analytics/cli-tools/gomanifest"
+	getGenerator := exec.Command(golang, "get", "-u", genRepo)
+	dependencyTree := exec.Command(golang, "run", genRepo, manifestDir, treePath, golang)
```

Comment:
@arajkumar I did try that, but that is Technically Not Possible.. 
`github.com/fabric8-analytics/cli-tools/gomanifest` is under package main... which by `go` terminology is parent package. Parent Packages can't be imported into any other packages.  

Work through is either we update package name https://github.com/fabric8-analytics/cli-tools/blob/main/gomanifest/build.go#L1 to something other than `main` which can potentially break existing functionality.


Ref https://stackoverflow.com/a/44421114/6782000

Reply:
The ask is not to import a main, but https://github.com/fabric8-analytics/cli-tools/tree/main/gomanifest/internal

Reply:
Internals are utility files for code written in https://github.com/fabric8-analytics/cli-tools/blob/main/gomanifest/build.go
So, are you looking forward to duplicating `build.go` just to refer internals from the new `build.go` ..?

Reply:
Basically you need to make use of the APIs exposed by github.com/fabric8-analytics/cli-tools/tree/main/gomanifest/internal to generate gomanifest. Possible reusable functionality shall be moved into internal from main pkg.

Reply:
"Reusing APIs exposed by Internals". 

That's exactly what the current `build.go` is doing! I believe we are circling back to rewrite `build.go` under a different package name..?

Reply:
My ask is simple, don't use cmd.Exec to generate manifest. In other ecosystems it is a must because you are relying on external tools to generate manifest. For golang we have a package which does the job.

Reply:
Okay, I'm rewriting `build.go` for CLI, under `golang` package

Reply:
@arajkumar There is an issue in importing `internals` too from `gomanifest` package in `golang` package because of `internals` being part of another executable.

> There are two types of packages. An executable package and a utility package. An executable package is your main application since you will be running it. A utility package is not self-executable, instead, it enhances the functionality of an executable package by providing utility functions and other important assets. 
Source: [Go Packages](https://medium.com/rungo/everything-you-need-to-know-about-packages-in-go-b8bac62b74cc )

![issue_importing_pkg](https://user-images.githubusercontent.com/21026016/105170966-b60a5900-5b43-11eb-934d-67c30b425899.png)

Reply:
We are again circling back to https://github.com/fabric8-analytics/cli-tools/pull/19#discussion_r560158804 
![gomanifest](https://user-images.githubusercontent.com/21026016/105283079-b9e6bb80-5bd5-11eb-8d4e-c4d2d6422e0e.png)

Reply:
Please refer https://github.com/fabric8-analytics/cli-tools/pull/20/commits/5947295273e37f85f1001758c7e9acee57423453

It has required changes.

Reply:
Ack.